modin/experimental/core/execution/native/implementations/hdk_on_native/dataframe/dataframe.py

```diff
@@ -1038,350 +1071,107 @@ def _union_all(
         HdkOnNativeDataframe
             The new frame.
         """
-        frames = [self] + other_modin_frames
-
-        # This is a special case, where we need to preserve the index of empty frames.
```

Comment:
How is this case handled now?

Reply:
[Here](https://github.com/modin-project/modin/pull/6251/files#diff-fc0f76c51d86e975ebb2af41cc109d92311368a9da5fc00a4b264f85d8bce737R950)